### ISSUE TITLE
NoSuchMethodError on //paste

### ISSUE THREAD

**Issue by SlimeDog** [InsightsPlugin/InsightsWorldEditExtension#8]
Spigot 3363-Spigot-f65de01-e36f759 (MC: 1.18.1)
WorldEdit version 7.2.9-SNAPSHOT+6024-9843a4f
Insights 6.9.1
InsightsWorldEditExtension 3.0.0

InsightsWorldEditExtension loads/enables successfully
```
[17:25:10] [Server thread/INFO]: [InsightsWorldEditExtension] Enabling InsightsWorldEditExtension v3.0.0
[17:25:10] [Server thread/INFO]: [InsightsWorldEditExtension] Successfully hooked into WorldEdit!
[17:25:10] [Server thread/INFO]: PM | The plugin has been loaded.
```
When I attempt to `//paste` a WE selection, it fails with an in-game error pointing to the console, which shows
```
[17:28:47] [Server thread/ERROR]: An unexpected error while handling a WorldEdit command
java.lang.NoSuchMethodError: 'int dev.frankheijden.insights.api.concurrent.storage.Distribution.count(java.util.function.Predicate)'
        at dev.frankheijden.insights.extensions.worldedit.InsightsExtentDelegate.onCommit(InsightsExtentDelegate.java:101) ~[?:?]
        at dev.frankheijden.insights.extensions.worldedit.InsightsAbstractDelegateExtent.commitBefore(InsightsAbstractDelegateExtent.java:50) ~[?:?]
        at com.sk89q.worldedit.extent.AbstractDelegateExtent.commit(AbstractDelegateExtent.java:128) ~[?:?]
        at com.sk89q.worldedit.extent.AbstractDelegateExtent.commit(AbstractDelegateExtent.java:129) ~[?:?]
        at com.sk89q.worldedit.EditSession.commit(EditSession.java:943) ~[?:?]
        at com.sk89q.worldedit.EditSession.internalFlushSession(EditSession.java:938) ~[?:?]
        at com.sk89q.worldedit.EditSession.setReorderMode(EditSession.java:375) ~[?:?]
        at com.sk89q.worldedit.EditSession.<init>(EditSession.java:299) ~[?:?]
        at com.sk89q.worldedit.EditSessionBuilder.build(EditSessionBuilder.java:143) ~[?:?]
        at com.sk89q.worldedit.LocalSession.createEditSession(LocalSession.java:1048) ~[?:?]
        at com.sk89q.worldedit.extension.platform.PlatformCommandManager.lambda$registerAlwaysInjectedValues$4(PlatformCommandManager.java:249) ~[?:?]
        at java.util.Optional.map(Optional.java:260) ~[?:?]
        at com.sk89q.worldedit.extension.platform.PlatformCommandManager.lambda$registerAlwaysInjectedValues$5(PlatformCommandManager.java:248) ~[?:?]
        at org.enginehub.piston.inject.MapBackedValueStore.lambda$injectedValue$0(MapBackedValueStore.java:56) ~[?:?]
        at java.util.Optional.flatMap(Optional.java:289) ~[?:?]
        at org.enginehub.piston.inject.MapBackedValueStore.injectedValue(MapBackedValueStore.java:56) ~[?:?]
        at org.enginehub.piston.inject.MergedValueAccess.lambda$injectedValue$0(MergedValueAccess.java:54) ~[?:?]
        at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197) ~[?:?]
        at java.util.Spliterators$ArraySpliterator.tryAdvance(Spliterators.java:1002) ~[?:?]
        at java.util.stream.ReferencePipeline.forEachWithCancel(ReferencePipeline.java:129) ~[?:?]
        at java.util.stream.AbstractPipeline.copyIntoWithCancel(AbstractPipeline.java:527) ~[?:?]
        at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:513) ~[?:?]
        at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499) ~[?:?]
        at java.util.stream.FindOps$FindOp.evaluateSequential(FindOps.java:150) ~[?:?]
        at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234) ~[?:?]
        at java.util.stream.ReferencePipeline.findFirst(ReferencePipeline.java:647) ~[?:?]
        at org.enginehub.piston.inject.MergedValueAccess.injectedValue(MergedValueAccess.java:57) ~[?:?]
        at org.enginehub.piston.inject.MemoizingValueAccess.injectedValue(MemoizingValueAccess.java:84) ~[?:?]
        at org.enginehub.piston.impl.CommandParametersImpl.injectedValue(CommandParametersImpl.java:109) ~[?:?]
        at org.enginehub.piston.inject.InjectedValueAccess.injectedValue(InjectedValueAccess.java:44) ~[?:?]
        at com.sk89q.worldedit.command.ClipboardCommandsRegistration.extract$editSession(ClipboardCommandsRegistration.java:324) ~[?:?]
        at com.sk89q.worldedit.command.ClipboardCommandsRegistration.cmd$_paste(ClipboardCommandsRegistration.java:260) ~[?:?]
        at org.enginehub.piston.CommandManager.execute(CommandManager.java:157) ~[?:?]
        at com.sk89q.worldedit.extension.platform.PlatformCommandManager.handleCommand(PlatformCommandManager.java:497) ~[?:?]
        at com.sk89q.worldedit.util.eventbus.MethodHandleEventHandler.dispatch(MethodHandleEventHandler.java:51) ~[?:?]
        at com.sk89q.worldedit.util.eventbus.EventHandler.handleEvent(EventHandler.java:73) ~[?:?]
        at com.sk89q.worldedit.util.eventbus.EventBus.dispatch(EventBus.java:193) ~[?:?]
        at com.sk89q.worldedit.util.eventbus.EventBus.post(EventBus.java:181) ~[?:?]
        at com.sk89q.worldedit.bukkit.WorldEditPlugin.onCommand(WorldEditPlugin.java:383) ~[?:?]
        at com.sk89q.bukkit.util.DynamicPluginCommand.execute(DynamicPluginCommand.java:55) ~[?:?]
        at org.bukkit.command.SimpleCommandMap.dispatch(SimpleCommandMap.java:149) ~[spigot-api-1.18.1-R0.1-SNAPSHOT.jar:?]
        at org.bukkit.craftbukkit.v1_18_R1.CraftServer.dispatchCommand(CraftServer.java:800) ~[spigot-1.18.1-R0.1-SNAPSHOT.jar:3363-Spigot-f65de01-e36f759]
        at net.minecraft.server.network.PlayerConnection.a(PlayerConnection.java:1938) ~[spigot-1.18.1-R0.1-SNAPSHOT.jar:3363-Spigot-f65de01-e36f759]
        at net.minecraft.server.network.PlayerConnection.a(PlayerConnection.java:1777) ~[spigot-1.18.1-R0.1-SNAPSHOT.jar:3363-Spigot-f65de01-e36f759]
        at net.minecraft.server.network.PlayerConnection.a(PlayerConnection.java:1758) ~[spigot-1.18.1-R0.1-SNAPSHOT.jar:3363-Spigot-f65de01-e36f759]
        at net.minecraft.network.protocol.game.PacketPlayInChat.a(PacketPlayInChat.java:46) ~[spigot-1.18.1-R0.1-SNAPSHOT.jar:3363-Spigot-f65de01-e36f759]
        at net.minecraft.network.protocol.game.PacketPlayInChat.a(PacketPlayInChat.java:1) ~[spigot-1.18.1-R0.1-SNAPSHOT.jar:3363-Spigot-f65de01-e36f759]
        at net.minecraft.network.protocol.PlayerConnectionUtils.lambda$0(PlayerConnectionUtils.java:30) ~[spigot-1.18.1-R0.1-SNAPSHOT.jar:3363-Spigot-f65de01-e36f759]
        at net.minecraft.server.TickTask.run(SourceFile:18) ~[spigot-1.18.1-R0.1-SNAPSHOT.jar:3363-Spigot-f65de01-e36f759]
        at net.minecraft.util.thread.IAsyncTaskHandler.c(SourceFile:151) ~[spigot-1.18.1-R0.1-SNAPSHOT.jar:3363-Spigot-f65de01-e36f759]
        at net.minecraft.util.thread.IAsyncTaskHandlerReentrant.c(SourceFile:23) ~[spigot-1.18.1-R0.1-SNAPSHOT.jar:3363-Spigot-f65de01-e36f759]
        at net.minecraft.server.MinecraftServer.b(MinecraftServer.java:1158) ~[spigot-1.18.1-R0.1-SNAPSHOT.jar:3363-Spigot-f65de01-e36f759]
        at net.minecraft.server.MinecraftServer.c(MinecraftServer.java:1) ~[spigot-1.18.1-R0.1-SNAPSHOT.jar:3363-Spigot-f65de01-e36f759]
        at net.minecraft.util.thread.IAsyncTaskHandler.y(SourceFile:125) ~[spigot-1.18.1-R0.1-SNAPSHOT.jar:3363-Spigot-f65de01-e36f759]
        at net.minecraft.server.MinecraftServer.bf(MinecraftServer.java:1137) ~[spigot-1.18.1-R0.1-SNAPSHOT.jar:3363-Spigot-f65de01-e36f759]
        at net.minecraft.server.MinecraftServer.y(MinecraftServer.java:1130) ~[spigot-1.18.1-R0.1-SNAPSHOT.jar:3363-Spigot-f65de01-e36f759]
        at net.minecraft.util.thread.IAsyncTaskHandler.c(SourceFile:134) ~[spigot-1.18.1-R0.1-SNAPSHOT.jar:3363-Spigot-f65de01-e36f759]
        at net.minecraft.server.MinecraftServer.x(MinecraftServer.java:1114) ~[spigot-1.18.1-R0.1-SNAPSHOT.jar:3363-Spigot-f65de01-e36f759]
        at net.minecraft.server.MinecraftServer.w(MinecraftServer.java:1038) ~[spigot-1.18.1-R0.1-SNAPSHOT.jar:3363-Spigot-f65de01-e36f759]
        at net.minecraft.server.MinecraftServer.lambda$0(MinecraftServer.java:304) ~[spigot-1.18.1-R0.1-SNAPSHOT.jar:3363-Spigot-f65de01-e36f759]
        at java.lang.Thread.run(Thread.java:833) [?:?]
```
If I `disable InsightsWorldEditExtension` I am able to `//paste` the WE selection.

**Comment by FrankHeijden** [comment]
Thanks for reporting, we’ll get this fixed somewhere this week :)

**Comment by FrankHeijden** [comment]
Fixed in v3.1.0, thanks!

**Comment by SlimeDog** [comment]
Confirmed fixed on Spigot 1.18.1-3363 with InsightsWorldEditExtension 3.1.0. Thanks.
